pandas/_libs/tslibs/fields.pyx

```diff
@@ -253,7 +253,8 @@ def get_start_end_field(
         # month of year. Other offsets use month, startingMonth as ending
         # month of year.
 
-        if freqstr[0:2]  in ["MS", "QS", "YS"]:
+        freq_name = freqstr.lstrip("B")[0:2]
+        if freq_name  in ["MS", "QS", "YS"]:
```

Comment:
sorry i meant that the argument `freqstr` in `get_start_end_field` needs renaming, because you're no longer passing in `freq.freqstr` but `freq.name`, so function argument (line 213) needs renaming

Reply:
thanks, it's clear now. I renamed the argument `freqstr` in `get_start_end_field`